sqlx-core/src/mysql/types/time.rs

```diff
@@ -101,21 +101,11 @@ impl<'r> Decode<'r, MySql> for Time {
                 decode_time(len - 5, buf)
             }
 
-            MySqlValueFormat::Text => {
-                let s = value.as_str()?;
-
-                // If there are less than 9 digits after the decimal point
-                // We need to zero-pad
-                // TODO: Ask [time] to add a parse % for less-than-fixed-9 nanos
-
-                let s = if s.len() < 20 {
-                    Cow::Owned(format!("{:0<19}", s))
-                } else {
-                    Cow::Borrowed(s)
-                };
-
-                Time::parse(&*s, "%H:%M:%S.%N").map_err(Into::into)
-            }
+            MySqlValueFormat::Text => Time::parse(
+                value.as_str()?,
+                &format_description::parse("[hour]:[minute]:[second].[subsecond]")?,
```

Comment:
`format_description::parse` could be replaced by the [format_description macro](https://docs.rs/time/0.3.2/time/macros/macro.format_description.html), which parses format descriptions at compile-time.

Reply:
I saw the macro but didn't look into it, does seem like the better option, thanks again.